utils/src/typename.rs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub trait TypeName {
+    fn typename_str() -> &'static str {
+        // This implementation is good enough, though it includes the full qualifiers of a typename, which may not be ideal
+        std::any::type_name::<Self>()
```

Comment:
I'm not sure about the default implementation. Yes, it is "good enough", but I guess that we want to provide a custom implementation for any type outside of the tests.

Reply:
What do you propose?

In its current form, we can choose to use the default or something else.

Reply:
Initially I thought that it almost never makes sense to use the default implementation, but now I get a slightly different idea: https://github.com/mintlayer/mintlayer-core/pull/449